### PR TITLE
Support week_ending_saturday for Druid.

### PR DESCRIPTION
**Pull request by bkyryliuk**
Implements: https://github.com/airbnb/caravel/issues/1495

In this change:

- refactored the granularity calculation to use `period` instead of `duration`
- added `week`, `week_starting_sunday`, `week_ending_saturday`, `month` granularities
- week_ending_saturday requires result post processing to offset the time to display the end of the week.

Tested:

- I've ran druid timeseries queries on couple examples for all data granularities - result look the same on dev and prod instances.

Reviewers:
* @mistercrunch 
* @ascott 
* @vera-liu

### CODE REVIEW

File: caravel/models.py [hunk 1]
```diff
@@ -1587,7 +1587,9 @@ def time_column_grains(self):
         return {
             "time_columns": [
                 'all', '5 seconds', '30 seconds', '1 minute',
-                '5 minutes', '1 hour', '6 hour', '1 day', '7 days'
+                '5 minutes', '1 hour', '6 hour', '1 day', '7 days',
```
Comment (bkyryliuk) [inline caravel/models.py]: @mistercrunch - are we using those ?

File: caravel/models.py [hunk 2]
```diff
@@ -1786,6 +1787,42 @@ def sync_to_db(cls, name, cluster):
...
+            'one day': 'P1D',
+            '1 day': 'P1D',
+            '7 days': 'P7D',
+            'week': 'P1W',
+            'week_starting_sunday': 'P1W',
+            'week_ending_saturday': 'P1W',
+            'month': 'P1M',
+        }
+
+        granularity = {'type': 'period', 'period': iso_8601_dict[period_name]}
```
Comment (mistercrunch) [inline caravel/models.py]: I think granularity is a free form select, which means you should use something like `iso_8601_dict.get(iso_8601_dict)` and fallback on passing in as `duration` if the entry is not in the dict

File: caravel/models.py [hunk 3]
```diff
@@ -1974,6 +2001,14 @@ def recursive_get_fields(_conf):
         cols += [col for col in groupby if col in df.columns]
         cols += [col for col in metrics if col in df.columns]
         df = df[cols]
+
+        time_offset = DruidDatasource.time_offset(granularity)
+        if 'timestamp' in df.columns and time_offset:
+            for i in range(len(df['timestamp'])):
```
Comment (mistercrunch) [inline caravel/models.py]: oh with pandas dataframe you always want to do set operations, something like `df.timestamp += timedelta(days=1)` should work. If you're doing it right, there's almost never a need to write a loop.

**Review by mistercrunch — CHANGES_REQUESTED** [review]

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
LGTM, though please think the implications of the timezone related changes. Could that change existing charts in ways that users may not expect or desire?

**Pull request merged**